core/src/main/scala/org/http4s/Message.scala

```diff
@@ -236,14 +237,14 @@ object Message {
   * @param body fs2.Stream[F, Byte] defining the body of the request
   * @param attributes Immutable Map used for carrying additional information in a type safe fashion
   */
-sealed abstract case class Request[F[_]](
-    method: Method = Method.GET,
-    uri: Uri = Uri(path = "/"),
-    httpVersion: HttpVersion = HttpVersion.`HTTP/1.1`,
-    headers: Headers = Headers.empty,
-    body: EntityBody[F] = EmptyBody,
-    attributes: Vault = Vault.empty
-) extends Message[F] {
+final class Request[F[_]](
+    val method: Method = Method.GET,
+    val uri: Uri = Uri(path = "/"),
+    val httpVersion: HttpVersion = HttpVersion.`HTTP/1.1`,
+    val headers: Headers = Headers.empty,
+    val body: EntityBody[F] = EmptyBody,
+    val attributes: Vault = Vault.empty
+) extends Message[F] with Product with Serializable {
```

Comment:
I'm not a huge fan of these interfaces, but:

* `Product` makes hash code easier and was there before.
* Spark folks may be depending on `Serializable`. I don't judge.